app/environments/brassbirmingham/brassbirmingham/envs/classes/board.py

```diff
@@ -129,10 +134,21 @@ def priceForIron(self, ironNeeded: int) -> int:
     """
 
     def areNetworked(
-        self, town1: Town | Building | TradePost, town2: Town | Building | TradePost
+        self, t1: Town | Building | TradePost, t2: Town | Building | TradePost
     ) -> bool:
-        q = [town1]
-        v = set(town1.id)
+        #town contains building
+        if type(t1) == Town and t2:
+            for buildLocation in t1.buildLocations:
+                if buildLocation.building and buildLocation.building.id == t2.id:
+                    return True
+        if type(t2) == Town and t1:
+            for buildLocation in t2.buildLocations:
+                if buildLocation.building and buildLocation.building.id == t1.id:
+                    return True
+                
+
+        q = [t1]
+        v = set(t1.id)
 
         while q:
             town = q.pop(0)  # bfs
```

Comment:
Not relevant to this PR really, but does this search work? We rely on `town.networks`, but `Building`s don't have networks?

Reply:
Actually L140-147 does the checking for buildings. Probably could be optimized/made cleaner but for now I think it's fine